poetry/poetry.py

```diff
@@ -83,7 +84,7 @@ def create(cls, cwd):  # type: (Path) -> Poetry
                 )
             )
 
-        local_config = TomlFile(poetry_file.as_posix()).read()
```

Comment:
If you override the `TomFile.read()` and apply the variable expansion there I think it would be better and could avoid future bugs where we call the `read()` without remembering to call `expand_environment_vars()` after.